src/autoscaler/routes/routes.go

```diff
@@ -59,88 +62,109 @@ const (
 	PublicApiHealthRouteName = "GetPublicApiHealth"
 )
 
-type AutoScalerRoute struct {
-	schedulerRoutes        *mux.Router
-	metricsCollectorRoutes *mux.Router
-	eventGeneratorRoutes   *mux.Router
-	scalingEngineRoutes    *mux.Router
-	metricsForwarderRoutes *mux.Router
-	apiOpenRoutes          *mux.Router
-	apiRoutes              *mux.Router
-	apiPolicyRoutes        *mux.Router
+type Router struct {
```

Comment:
@silvestre the mux.routers created before are now a single mux.router, this package now implements some instance method that generate subrouter from the main instance when we need to register the same routes with 2 different middleware (eg: one with xfcc-auth and another without it). We need different instances of the router with the same register routes.  The changes in this file were necessary for two reasons. 1. to support any component that consumes this package as a singleton router (autoScalerRouteInstance in this file) and to be consumed by a public API which now offers a cfServer with the same routes as the mtlsServer but with different set of middlewares.